qiskit_nature/mappers/second_quantization/logarithmic_mapper.py

```diff
@@ -0,0 +1,199 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Logarithmic Mapper."""
+
+import operator
+
+from fractions import Fraction
+from functools import reduce
+from typing import List, Union
+
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import SparsePauliOp, Operator
+from qiskit_nature.operators.second_quantization import SpinOp
+from .spin_mapper import SpinMapper
+
+
+class LogarithmicMapper(SpinMapper):
+    """A mapper for Logarithmic spin-to-qubit mapping [1].
+
+
+    [1]: Mathis, S. V., Gauglielmo, M., & Ivano, T. (2020).
+         Toward scalable simulations of Lattice Gauge Theories on quantum computer.
+         `arXiv:2005.10271 <https://arxiv.org/abs/2005.10271>`_
+    """
+
+    def __init__(self, embed_padding: float = 1, embed_location: str = "upper") -> None:
+        """
+        Args:
+            embed_padding:
+                The value of the diagonal elements of the upper left  block of the embedded matrix.
+            embed_location:
+                Must be one of ['upper', 'lower']. This parameters sets whether
+                the given matrix is embedded in the upper left hand corner or
+                the lower right hand corner of the larger matrix.
+        """
+        super().__init__(allows_two_qubit_reduction=False)
+        self._embed_padding = embed_padding
+        self._embed_location = embed_location
+
+    def map(self, second_q_op: SpinOp) -> PauliSumOp:
+        """Map spins to qubits using the Logarithmic encoding.
+        In this local encoding transformation each individual spin S system is represented via
+        the lowest lying 2S+1 states in a qubit system with the minimal number of qubits needed to
+        represent >= 2S+1 distinct states.
+
+        Args:
+            second_q_op: Spins mapped to qubits.
+
+        Returns:
+            Qubit operators generated by the Logarithmic encoding
+        """
+        qubit_ops_list: List[PauliSumOp] = []
+
+        # get logarithmic encoding of the general spin matrices.
+        spinx, spiny, spinz, identity = self._logarithmic_encoding(second_q_op.spin)
+
+        for idx, (_, coeff) in enumerate(second_q_op.to_list()):
+
+            operatorlist: List[PauliSumOp] = []
+
+            for n_x, n_y, n_z in zip(second_q_op.x[idx], second_q_op.y[idx], second_q_op.z[idx]):
+
+                operator_on_spin_i: List[PauliSumOp] = []
+
+                if n_x > 0:
+                    operator_on_spin_i.append(reduce(operator.matmul, [spinx] * int(n_x)))
```

Comment:
I do not understand the need for the `reduce()` call. Can you please explain why this is necessary or update the code if it is not.

Reply:
Oh I overlooked the fact that the list * integer multiplication will chain the list. Yeh this makes sense now, thanks! :+1: